src/main/java/org/opensrp/web/rest/LocationResource.java

```diff
@@ -387,6 +399,48 @@ public void setLocationService(PhysicalLocationService locationService) {
 		return new ResponseEntity<>(identifiers, RestUtils.getJSONUTF8Headers(), HttpStatus.OK);
 
 	}
+	
+	@RequestMapping(value = "/search-location", method = RequestMethod.GET, produces = { MediaType.APPLICATION_JSON_VALUE })
+	@ResponseBody
+	public ResponseEntity<String> searchLocations(@RequestParam(value = "locationTagId", required = false) Long locationTagId,
+	                                              @RequestParam(value = "name", required = false) String name,
+	                                              @RequestParam(value = "parentId", required = false) Long parentId,
+	                                              @RequestParam(value = "status", required = false) String status,
+	                                              @RequestParam(value = "orderByFieldName", required = false) String orderByFieldName,
+	                                              @RequestParam(value = "orderByType", required = false) String orderByType,
+	                                              @RequestParam(value = "pageSize", required = false) Integer pageSize,
+	                                              @RequestParam(value = "pageNumber", required = false) Integer pageNumber)
+	    throws JsonProcessingException {
+		LocationSyncBean locationSyncBean = new LocationSyncBean();
+		LocationSearchBean locationSearchBean = new LocationSearchBean();
+		
+		try {
+			locationSearchBean.setName(name);
+			locationSearchBean.setLocationTagId(locationTagId);
+			locationSearchBean.setOrderByFieldName(orderByFieldName);
+			if (orderByType != null) {
+				locationSearchBean.setOrderByType(OrderByType.valueOf(orderByType));
+			}
+			locationSearchBean.setPageNumber(pageNumber);
+			locationSearchBean.setPageSize(pageSize);
+			locationSearchBean.setStatus(status);
+			locationSearchBean.setParentId(parentId);
+			List<CustomPhysicalLocation> locations = locationService.searchLocations(locationSearchBean);
+			
+			locationSyncBean.setCustomLocations(locations);
+			int total = 0;
+			if (pageNumber != null && pageNumber == 1) {
+				total = locationService.countSearchLocations(locationSearchBean);
+			}
+			locationSyncBean.setTotal(total);
+		}
+		catch (IllegalArgumentException e) {
+			return new ResponseEntity<String>(e.getMessage(), HttpStatus.BAD_REQUEST);
+		}
+		return new ResponseEntity<>(objectMapper.writeValueAsString(locationSyncBean), RestUtils.getJSONUTF8Headers(),
```

Comment:
This API is using a gson serializer and not yet migrated to Jackson. Kindly use the gson serializer on the API so that the API retains consistency.  This will moved later for all the endpoints